### ISSUE TITLE
Documentation : Need explanation on how the sound is produced

### ISSUE THREAD

**Issue by ParikshithMohan** [224-1/psilib-html5-theremin#10]
Cant understand these ->

;; Define Audio Context, Oscillator & Gain Node
(defonce context (js/window.AudioContext.))
(defonce osc (.createOscillator context))
(defonce amp (.createGain context))

;; Create Wiring
(. osc connect amp)
(. amp connect (.-destination context))

;; Initialise Oscillator & Gain
(set! (.-type osc) (:osc @inst-state))
(set! (.-value (.-detune osc)) (:detune @inst-state))
(set! (.-value (.-gain amp)) (:gain @inst-state))

**Comment by kattakavya** [comment]
what is use of creating gain context ? will max,min gain will all come together?

**Comment by ParikshithMohan** [comment]
Can somebody explain over a discord call?

**Comment by sahabaishakhi** [comment]
yes, let's get on a call tomorrow about this. I can explain

**Comment by azureus89** [comment]
@sahabaishakhi @ParikshithMohan @kattakavya Let me know if from baishakhis explanation you had understood better.
@sahabaishakhi Im not entirely convinced you have understood the code yourself based on the pull request submitted, the solution you had also was not optimal.

Please explain your understnding of the problem, based on which I will decide on how much documentation is necessary.

**Comment by sahabaishakhi** [comment]
I explained how the audiocontext is an object used which is an audioprocessing graph with audio nodes on which certain methods are used to generate sounds. I explained side by side the javascript code and how clojure script is used to appropriate the same. for example, `var context = new AudioContext()` is the same as `(defonce context (js/window.AudioContext.))` and `var osc = context.createOscillator()` and `(defonce osc (.createOscillator context))` are similar. where createOscillator or createGain are methods to define the oscillator and amplifier components. after that explained about the reagent atom and the components described within the map, namely `osc` to change the 'type' of the oscillator `detune` whose  documentation I had to refer to find out that it's number of oscillations for music intervals. within that documentation there's an example of a violent theremin whose detune value is set as 100. also `amp` whose gain is a certain value. I explained what gain really represents which is loudness.. and how the atom's values are derefed and set by using web audio api properties like 'type' and detune and gain's value. I went on the explain the if clause to kavya where the status of the oscillator is again derefed and incremented after applying the code in do block. the oscillator is started using (. osc start 0) where 0 is the time reference in milliseconds at which we are to start producing the sound. referred to the resume method in web audio API documentation to resume the progression of time in a previously suspended session. finally I explained the wiring part where the components oscillator and the amplifier are supposed to be connnected and furthermore the amplifier is to be connected to the audiocontext's destination which is the output for the sound
